numpy/lib/nanfunctions.py

```diff
@@ -1109,6 +1128,78 @@ def nanpercentile(a, q, axis=None, out=None, overwrite_input=False,
     the median if ``q=50``, the same as the minimum if ``q=0`` and the
     same as the maximum if ``q=100``.
 
+    In the simplest case, where all weights are integers,
```

Comment:
I don't like how this massive block is repeated in four places. Perhaps put it in just quantile, and have the other three point to it for more details about weighting?

Reply:
I'm all for it.  I think there was an email thread advocating for redundancy between the docstrings of `quantile`, `percentile`, `nanquantile`, and `nanpercentile`.  But perhaps the details here don't need to be.